hadoop-ozone/ozone-manager/src/main/java/org/apache/hadoop/ozone/om/request/key/OMKeysRenameRequest.java

```diff
@@ -0,0 +1,279 @@
+              fromKey);
+          continue;
+        }
+
+        fromKeyValue.setUpdateID(trxnLogIndex, ozoneManager.isRatisEnabled());
+
+        fromKeyValue.setKeyName(toKeyName);
+
+        //Set modification time
+        fromKeyValue.setModificationTime(Time.now());
+
+        acquiredLock =
+            omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
+                volumeName, bucketName);
+        // Add to cache.
+        // fromKey should be deleted, toKey should be added with newly updated
+        // omKeyInfo.
+        Table<String, OmKeyInfo> keyTable = omMetadataManager.getKeyTable();
+        keyTable.addCacheEntry(new CacheKey<>(fromKey),
+            new CacheValue<>(Optional.absent(), trxnLogIndex));
+        keyTable.addCacheEntry(new CacheKey<>(toKey),
+            new CacheValue<>(Optional.of(fromKeyValue), trxnLogIndex));
+        renamedKeys.put(fromKeyName, toKeyName);
+        fromKeyAndToKeyInfo.put(fromKeyName, fromKeyValue);
+        if (acquiredLock) {
+          omMetadataManager.getLock().releaseWriteLock(BUCKET_LOCK, volumeName,
+              bucketName);
+          acquiredLock = false;
+        }
+      }
+
+      acquiredLock =
+          omMetadataManager.getLock().acquireWriteLock(BUCKET_LOCK,
+              volumeName, bucketName);
+      OmRenameKeys newOmRenameKeys =
+          new OmRenameKeys(volumeName, bucketName, null, fromKeyAndToKeyInfo);
+      omClientResponse = new OMKeysRenameResponse(omResponse
+          .setRenameKeysResponse(RenameKeysResponse.newBuilder()
+              .setStatus(renameStatus)
+              .addAllUnRenamedKeys(unRenamedKeys))
+          .setStatus(renameStatus ? OK : PARTIAL_RENAME)
+          .setSuccess(renameStatus).build(),
+          newOmRenameKeys);
+
+      result = Result.SUCCESS;
+    } catch (IOException ex) {
+      result = Result.FAILURE;
+      exception = ex;
+      createErrorOMResponse(omResponse, ex);
+
+      omResponse.setRenameKeysResponse(RenameKeysResponse.newBuilder()
+          .setStatus(renameStatus).addAllUnRenamedKeys(unRenamedKeys).build());
+      omClientResponse = new OMKeysRenameResponse(omResponse.build());
+
+    } finally {
+      addResponseToDoubleBuffer(trxnLogIndex, omClientResponse,
+          omDoubleBufferHelper);
+      if (acquiredLock) {
+        omMetadataManager.getLock().releaseWriteLock(BUCKET_LOCK, volumeName,
+            bucketName);
+      }
+    }
+
+    auditMap = buildAuditMap(auditMap, renamedKeys, unRenamedKeys);
+    auditLog(auditLogger, buildAuditMessage(OMAction.RENAME_KEYS, auditMap,
+        exception, getOmRequest().getUserInfo()));
+
+    switch (result) {
+    case SUCCESS:
+      LOG.debug("Rename Keys is successfully completed for auditMap:{}.",
+          auditMap.toString());
+      break;
+    case FAILURE:
+      ozoneManager.getMetrics().incNumKeyRenameFails();
+      LOG.error("Rename keys failed for auditMap:{}.", auditMap.toString());
```

Comment:
Avoid eager `toString()` call in logging.

```suggestion
      LOG.error("Rename keys failed for auditMap:{}.", auditMap);
```